Modules/Filtering/Convolution/include/itkFFTConvolutionImageFilter.hxx

```diff
@@ -126,47 +147,160 @@ FFTConvolutionImageFilter<TInputImage, TKernelImage, TOutputImage, TInternalPrec
+  InputRegionType  inputRequestedRegion = input->GetRequestedRegion();
+  InputSizeType    inputRequestedSize = inputRequestedRegion.GetSize();
+  InputIndexType   inputRequestedIndex = inputRequestedRegion.GetIndex();
+  OutputRegionType outputRequestedRegion = this->GetOutput()->GetRequestedRegion();
+  OutputSizeType   outputRequestedSize = outputRequestedRegion.GetSize();
+  OutputIndexType  outputRequestedIndex = outputRequestedRegion.GetIndex();
+
+  // Pad the input image such that the requested region, expanded by
+  // twice the kernel radius, lies entirely within the buffered region.
+  // If the requested region is at least the kernel radius away from
+  // all boundaries of the buffered input region then padding is skipped.
 
   using InputPadFilterType = PadImageFilter<InputImageType, InputImageType>;
-  auto inputPadder = InputPadFilterType::New();
-  inputPadder->SetBoundaryCondition(this->GetBoundaryCondition());
+  using PadSizeType = typename InputPadFilterType::SizeType;
 
-  InputSizeType inputLowerBound = this->GetPadLowerBound();
-  inputPadder->SetPadLowerBound(inputLowerBound);
+  PadSizeType    lowerPad, upperPad;
+  bool           needsKernelPadding = false;
+  KernelSizeType kernelRadius = this->GetKernelRadius();
+  for (unsigned int dim = 0; dim < ImageDimension; ++dim)
+  {
+    signed int lower = static_cast<signed int>(kernelRadius[dim]) - (inputRequestedIndex[dim] - inputLargestIndex[dim]);
+    signed int upper =
+      static_cast<signed int>(kernelRadius[dim]) -
+      ((inputLargestIndex[dim] + inputLargestSize[dim]) - (inputRequestedIndex[dim] + inputRequestedSize[dim]));
+    lowerPad[dim] = (lower > 0 ? lower : 0);
+    upperPad[dim] = (upper > 0 ? upper : 0);
+
+    needsKernelPadding = (lower > 0 || upper > 0);
+  }
 
-  InputSizeType inputUpperBound;
-  for (unsigned int i = 0; i < ImageDimension; ++i)
+  const InputImageType * kernelPaddedInput;
+  if (needsKernelPadding)
+  {
+    auto inputPadder = InputPadFilterType::New();
+    inputPadder->SetBoundaryCondition(this->GetBoundaryCondition());
+    inputPadder->SetPadLowerBound(lowerPad);
+    inputPadder->SetPadUpperBound(upperPad);
+    inputPadder->SetNumberOfWorkUnits(this->GetNumberOfWorkUnits());
+    inputPadder->SetInput(input);
+    inputPadder->ReleaseDataFlagOn();
+    progress->RegisterInternalFilter(inputPadder, 0.2f * progressWeight);
+    remainingProgress -= 0.2f;
+    inputPadder->Update();
+    kernelPaddedInput = inputPadder->GetOutput();
+  }
+  else
+  {
+    kernelPaddedInput = input;
+  }
+
+  // Crop to region of interest to minimize FFT requested region size.
+  // This blocks forward FFT from requesting the entire input image
+  // when only a subregion (output requested region + kernel radius padding)
+  // is required for convolution to succeed.
+  // TODO: Improve readability by keeping the output index as-is rather than resetting to 0, if the filter allows it
```

Comment:
Isn't this being done on lines 235-250?

Reply:
Yes, I haven't yet looked into whether there is a way to preserve the image index with `itk::RegionOfInterestImageFilter` without adding an additional `itk::ChangeInformationImageFilter` to the pipeline. The performance impact is negligible either way, would be nice if we can do this by setting a parameter though.